packages/grub2/config/grub.cfg

```diff
@@ -15,37 +29,50 @@ else
   set default="${saved_entry}"
 fi
 
-set fallback="0 1 2"
+## Display a default menu entry if set
+if [ "${default_menu_entry}" ]; then
+  set display_name="${default_menu_entry}"
+else
+  set display_name="cOS"
+fi
+
+## Set a default fallback if set
+if [ "${default_fallback}" ]; then
+  set fallback="${default_fallback}"
+else
+  set fallback="0 1 2"
+fi
+
 set gfxmode=auto
 set gfxpayload=keep
 insmod all_video
 insmod gfxterm
 insmod loopback
 insmod squash4
 
-menuentry "cOS" --id cos {
+menuentry "${display_name}" --id cos {
   search.fs_label COS_STATE root
   set img=/cOS/active.img
   set label=COS_ACTIVE
   loopback loop0 /$img
   set root=($root)
   source (loop0)/etc/cos/bootargs.cfg
-  linux (loop0)$kernel $kernelcmd
+  linux (loop0)$kernel $kernelcmd ${extra_cmdline} ${extra_active_cmdline}
```

Comment:
Not directly related, but should we just load everything, including kernelcmd from env files directly so the whole thing is overwritable?

Reply:
the kernelcmd comes from the derivative itself already as an env file, so the derivative have full control over the boot commands. However, there is no explicit way for users to pin specific flags, while now you could:

For all:
```
grub2-editenv /oem/grubenv set extra_cmdline=...
``` 

For the single entry, e.g. active:
```
grub2-editenv /oem/grubenv set extra_active_cmdline=...
```